### PR TITLE
Feature/add graphql codegen

### PR DESCRIPTION
**Pull request by riccardo-vettore**
- Added graphql codegen dependencies
- Migrated products queries

### CODE REVIEW

File: codegen.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+overwrite: true
+schema:
+  - 'https://demo.vendure.io/shop-api'
```
Comment (gioboa) [inline codegen.yml:3]: 👍 now we are using this vendure endpoint

```suggestion
  - 'https://readonlydemo.vendure.io/shop-api'
```

File: src/providers/products/products.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
+import gql from 'graphql-tag';
+import { sdk } from '~/graphql-wrapper';
+import { SearchInput } from '~/generated/graphql';
+
+export const search = async (searchInput: SearchInput) => {
```
Comment (gioboa) [inline src/providers/products/products.ts:5]: are these providers files generated automatically?

File: codegen.yml (hunk 1 above)
Comment (riccardo-vettore) [reply codegen.yml:3]: 👍

File: src/providers/products/products.ts (hunk 2 above)
Comment (riccardo-vettore) [reply src/providers/products/products.ts:5]: Providers files aren't generated automatically

File: src/routes/collections/[...slug]/index.tsx [hunk 3]
```diff
@@ -11,32 +11,29 @@ import CollectionCard from '~/components/collection-card/CollectionCard';
...
 } from '~/utils';
-import { execute } from '~/utils/api';
+import { getCollectionBySlug } from '~/providers/collections/collections';
+import { searchQueryWithCollectionSlug, searchQueryWithTerm } from '~/providers/products/products';
+import { Collection, SearchResponse } from '~/generated/graphql';
 
 export const useCollectionLoader = routeLoader$(async ({ params }) => {
-	const { collection } = await execute<{ collection: Collection }>(getCollectionQuery(params.slug));
-	return collection;
+	const { collection } = await getCollectionBySlug(params.slug);
```
[the remark this answers is not in the document]
Comment (riccardo-vettore) [reply src/routes/collections/[...slug]/index.tsx:27]: Could this be a solution?
```html
```

File: src/providers/products/products.ts (hunk 2 above)
Comment (gioboa) [reply src/providers/products/products.ts:5]: 💔

File: codegen.yml (hunk 1 above)
Comment (riccardo-vettore) [reply codegen.yml]: I have pushed a possible solution. Let me know if it's ok 👍

**Review by gioboa — APPROVED** [review]
Looks great 👍

**Pull request merged**